File: src/main/java/transportsystem/model/Engine.java
package transportsystem.model;

import javax.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.util.List;

@Entity
@Table(name="engine")
public class Engine {

    @Id
    @Column
    @GeneratedValue(strategy = GenerationType.IDENTITY)
	@Getter
	@Setter
	private Integer id;

    @Column
    @Getter
    @Setter
	private String name;

    @Column
    @Getter
    @Setter
	private Double volume;

    @Column
    @Getter
    @Setter
	private String fuel;

    @Getter
    @Setter
    @OneToMany(mappedBy = "engine", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private List<TransportModel> transportmodels;

}

File: frontend/src/models/engine.js
export default class Engine {
    constructor(data) {
        if (data === undefined) {
            this.name = '';
            this.volume = '';
            this.fuel = ''
        }
        else {
            this.id = data.id;
            this.name = data.name;
            this.volume = data.volume;
            this.fuel = data.fuel;
        }
    }
}